plugins/flytekit-greatexpectations/flytekitplugins/great_expectations/task.py

```diff
@@ -0,0 +1,275 @@
+        batch_identifiers: identifiers to identify the data batch
+        batch_spec_passthrough: reader method if your file doesn't have an extension
+    """
+
+    data_connector_query: Optional[Dict[str, Any]] = None
+    runtime_parameters: Optional[Dict[str, Any]] = None
+    batch_identifiers: Optional[Dict[str, str]] = None
+    batch_spec_passthrough: Optional[Dict[str, Any]] = None
+
+
+class GreatExpectationsTask(PythonInstanceTask[BatchRequestConfig]):
+    """
+    This task can be used to validate your data.
+
+    Args:
+        name: name of the task
+        datasource_name: tell where your data lives and how to get it
+        expectation_suite_name: suite which consists of the data expectations
+        data_connector_name: connector to identify data batches
+        inputs: inputs to pass to the execute() method
+        data_asset_name: name of the data asset (to be used for RuntimeBatchRequest)
+        local_file_path: dataset file path useful for FlyteFile and FlyteSchema
+        checkpoint_params: optional SimpleCheckpoint parameters
+        task_config: batchrequest config
+        context_root_dir: directory in which GreatExpectations' configuration resides
+
+    TODO: Connect Data Docs to Flyte Console.
+    """
+
+    _TASK_TYPE = "great_expectations"
+    _RUNTIME_VAR_NAME = "runtime"
+
+    def __init__(
+        self,
+        name: str,
+        datasource_name: str,
+        expectation_suite_name: str,
+        data_connector_name: str,
+        inputs: Dict[str, Type],
+        data_asset_name: Optional[str] = None,
+        outputs: Optional[Dict[str, Type]] = None,
+        local_file_path: Optional[str] = None,
+        checkpoint_params: Optional[Dict[str, Union[str, List[str]]]] = None,
+        task_config: BatchRequestConfig = None,
+        context_root_dir: str = "./great_expectations",
+        **kwargs,
+    ):
+        self._datasource_name = datasource_name
+        self._data_connector_name = data_connector_name
+        self._expectation_suite_name = expectation_suite_name
+        self._batch_request_config = task_config
+        self._context_root_dir = context_root_dir
+        self._data_asset_name = data_asset_name
+        """
+        local_file_path is a must in two scenarios:
+        * When using FlyteSchema
+        * When using FlyteFile for remote paths
+        This is because base directory which has the dataset file 'must' be given in GreatExpectations' config file
+        """
+        self._local_file_path = local_file_path
+        self._checkpoint_params = checkpoint_params
+
+        outputs = {"result": Dict[Any, Any]}
+
+        super(GreatExpectationsTask, self).__init__(
+            name=name,
+            task_config=task_config,
+            task_type=self._TASK_TYPE,
+            interface=Interface(inputs=inputs, outputs=outputs),
+            **kwargs,
+        )
+
+    def execute(self, **kwargs) -> Any:
```

Comment:
I think the method is too long, can we split the various `if` conditions into sub-functions. Makes it easier to read

Reply:
I split FlyteFile and FlyteSchema into two methods in both task.py and schema.py files.